proxy/proxy.go

```diff
@@ -169,23 +224,56 @@ func mapRequest(r *http.Request, rt *routing.Route, host string) (*http.Request,
 	return rr, nil
 }
 
-// Creates a proxy. It expects a routing instance that is used to match
-// the incoming requests to routes. If the 'insecure' parameter is true, the
-// proxy skips the TLS verification for the requests made to the
-// route backends. It accepts an optional list of priority routes to
-// be used for matching before the general lookup tree.
-func New(r *routing.Routing, options Options, pr ...PriorityRoute) http.Handler {
-	tr := &http.Transport{}
-	if options.Insecure() {
+// Deprecated, see NewProxy and ProxyOptions instead.
+func New(r *routing.Routing, options Options, pr ...PriorityRoute) *Proxy {
+	return NewProxy(ProxyOptions{
+		Routing:              r,
+		Flags:                ProxyFlags(options),
+		PriorityRoutes:       pr,
+		CloseIdleConnsPeriod: -time.Second})
+}
+
+// Creates a proxy.
+func NewProxy(o ProxyOptions) *Proxy {
```

Comment:
I see the dilemma... to avoid breaking the API, we now get `proxy.NewProxy(...)` ... Would it be too shocking to export only this one under skipper pkg? I understand this means having the factory method further away from the implementation but it actually only uses exported stuff. `skipper.NewProxy(...)` would seem more idiomatic...